### PR TITLE
BRS-643: Role assignment for Sub-areas in A&R

### PR DESCRIPTION
**Pull request by marklise**
This changes how roles are worked in A&R.  Filtering based on Park and/or Subarea for all api calls. Removed `POST park` as it's unused at the moment, but may come back soon.  If not, a new ticket should be put into the backlog to address the complete removal.

NB: This requires a migration to be run (20220621153946-parkAndSubAreaRoles) as well as this PR to be pulled down during testing of this back-end change:  https://github.com/bcgov/bcparks-ar-admin/pull/144

### CODE REVIEW

File: lambda/park/GET/index.js [hunk 1]
```diff
@@ -23,6 +33,25 @@ exports.handler = async (event, context) => {
...
       } while (typeof parkData.LastEvaluatedKey !== "undefined");
+
+      if (permissionObject.isAdmin) {
+        // Sysadmin, they get it all
+        logger.debug("**Sysadmin**")
+      } else {
+        // Some other authenticated role
+        logger.debug("**Some other authenticated person with attendance-and-revenue roles**")
+        // logger.debug("results1:", results);
+        // logger.debug("permissionObject.roles:", permissionObject.roles);
```
Comment (cameronpettit) [inline lambda/park/GET/index.js:44]: nit - remove commented code

File: lambda/permissionUtil.js [hunk 2]
```diff
@@ -1,37 +1,41 @@
 const jwt = require('jsonwebtoken');
 const jwksClient = require('jwks-rsa');
-const SSO_ISSUER = process.env.SSO_ISSUER || 'https://oidc.gov.bc.ca/auth/realms/g7v0xlf4';
-const SSO_JWKSURI = process.env.SSO_JWKSURI || 'https://oidc.gov.bc.ca/auth/realms/g7v0xlf4/protocol/openid-connect/certs';
+const SSO_ISSUER = process.env.SSO_ISSUER || 'https://dev.oidc.gov.bc.ca/auth/realms/g7v0xlf4';
+const SSO_JWKSURI = process.env.SSO_JWKSURI || 'https://dev.oidc.gov.bc.ca/auth/realms/g7v0xlf4/protocol/openid-connect/certs';
```
Comment (cameronpettit) [inline lambda/permissionUtil.js:4]: Do we not want this to default to PROD oidc?

File: migrations/20220621153946-parkAndSubAreaRoles.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+  try {
+    const parks = await getParks();
+
+    console.log("parks:", parks);
+
+    for(const park of parks) {
+      await processPark(park);
+    }
+  } catch (err) {
+    logger.error(err);
```
Comment (cameronpettit) [inline migrations/20220621153946-parkAndSubAreaRoles.js:89]: `logger` is not declared in this migration script.

Comment (marklise) [reply migrations/20220621153946-parkAndSubAreaRoles.js]: Fixed in 0163feb

File: lambda/permissionUtil.js (hunk 2 above)
Comment (marklise) [reply lambda/permissionUtil.js:4]: No - default to dev, PROD will have SSO_* set in the lambda.

File: lambda/park/GET/index.js (hunk 1 above)
Comment (marklise) [reply lambda/park/GET/index.js:44]: Fixed in 0163feb

**Review by cameronpettit — DISMISSED** [review]

**Review by meredithom — APPROVED** [review]

**Review by cameronpettit — APPROVED** [review]

**Pull request merged**